include/moq/detail/transport.h

```diff
@@ -116,7 +116,7 @@ namespace moq {
          * @param connection_handle         Connection ID to send subscribe
          * @param track_handler             Track handler to use for track related functions and callbacks
          */
-        void UnsubscribeTrack(ConnectionHandle connection_handle, std::shared_ptr<SubscribeTrackHandler> track_handler);
+        void UnsubscribeTrack(ConnectionHandle connection_handle, const std::shared_ptr<SubscribeTrackHandler>& track_handler);
```

Comment:
These were recommended by clang-tidy since we never use the copy of the shared ptrs. If this is the wrong assumption then let's talk about it.

Reply:
It was moved before, but this works as well. We should be consistent here because the school of thought was that we should pass as a copy/ref count and then move within the method if that is the intent.

Reply:
Yeah, but since we never moved then there's no point in copying the pointer, so pass by ref makes it clearer that we are not necessarily copying it (which we aren't).

Reply:
This is another situation where this function doesn't involve lifetime, could it not just be a const ref?

Reply:
I suppose the `Un*` methods don't need shared pointer, but if we update one they all should be updated to be consistent.

Reply:
Very possibly it could just be a const&. I'm not sure if changing it from shared_ptr makes it inconsistent with the PublishTrack and SubscribeTrack, which both use shared_ptrs (which there is no moving away from). Is it worth needing to dereference our shared_ptrs to use these Un* methods?